### PR TITLE
Updates to extensions NS management

### PR DESCRIPTION
**Pull request by ddavidov**
- Important fixes

### CODE REVIEW

File: zoolanders/Framework/Container/Container.php [hunk 1]
```diff
@@ -123,14 +123,14 @@ public function registerExtension($name, $path = null, $namespace = null)
...
-            $path = JPATH_ADMINISTRATOR . '/components/con_' . strtolower($componentName);
+            $path = JPATH_ADMINISTRATOR . '/components/' . strtolower($componentName);
         }
 
         if (!$namespace) {
             $namespace = Container::ROOT_NAMESPACE . ucfirst(strtolower($name)) . '\\';
         }
 
-        $this->registeredExtensions[strtolower($name)][] = $namespace;
+        $this->registeredExtensions['com_' . strtolower($name)][] = $namespace;
```
Comment (Skullbock) [inline zoolanders/Framework/Container/Container.php]: don't do this, the name has to remain without the com_ . Ex: zoolingual

File: zoolanders/Framework/Dispatcher/Dispatcher.php [hunk 2]
```diff
@@ -122,6 +122,7 @@ public function dispatch($default = null)
 
                 if ($response instanceof ResponseInterface) {
                     $response->send();
+                    break;
```
Comment (Skullbock) [inline zoolanders/Framework/Dispatcher/Dispatcher.php]: use "return"

Comment (Skullbock) [inline zoolanders/Framework/Dispatcher/Dispatcher.php]: Actually, cancel that. There is the event later

Comment (Skullbock) [inline zoolanders/Framework/Dispatcher/Dispatcher.php]: so break is good

File: zoolanders/Framework/Service/Factory.php [hunk 3]
```diff
@@ -44,8 +44,20 @@ public function view($input, $config = [])
     {
         $type = $input->isAjax() ? 'Json' : 'Html';
         $name = $config['name'];
+        $viewClass = '';
 
-        $viewClass = $this->container->environment->getRootNamespace() . 'View\\' . ucfirst($name) . '\\' . $type;
+        $component = $this->container->system->application->input->getCmd('option');
```
Comment (Skullbock) [inline zoolanders/Framework/Service/Factory.php]: Create a helper method in the environment service to get the current extension (which is stored somewhere in the dispatcher i think)

File: zoolanders/Framework/Container/Container.php (hunk 1 above)
Comment (ddavidov) [inline zoolanders/Framework/Container/Container.php]: If so, when we are retrieving getRegisteredNamespaces, we need to specify ext name without prefixes. If we take that from request we would need to cleanup option value from com_, plg_ etc.

File: zoolanders/Framework/Dispatcher/Dispatcher.php (hunk 2 above)
Comment (ddavidov) [inline zoolanders/Framework/Dispatcher/Dispatcher.php]: +

File: zoolanders/Framework/Service/Factory.php (hunk 3 above)
Comment (ddavidov) [inline zoolanders/Framework/Service/Factory.php]: get current exception without prefix, right?

File: zoolanders/Framework/Container/Container.php (hunk 1 above)
Comment (Skullbock) [inline zoolanders/Framework/Container/Container.php]: correct, we definetly should. the idea is that if the ext is grouped into component + plugins, you wouldn't have to register 5 extensions if you want them to share the namespace. Just register multiple paths for the same namespace

**Pull request merged**